### PR TITLE
Updated dependencies (`typescript`)

### PR DESCRIPTION
**Pull request by bwp91**
Updated `typescript` to the latest version, and the `@typescript-eslint/` dependencies too.

---

### Changed

- Updated dependencies (`typescript`)

### CODE REVIEW

**Review by donavanbecker — DISMISSED** [review]

File: src/internal-types.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable @typescript-eslint/no-duplicate-enum-values */
```
Comment (bauer-andreas) [inline src/internal-types.ts]: Can we use `esline-disable-next` and attach them to the specific line that triggers the check, if it makes sense here  (e.g., like here https://github.com/homebridge/HAP-NodeJS/blob/110a6924d6c595efb562032a005eca887a7fd961/src/types/dbus-native.d.ts#L10).
This allows us to have a minimal surface where rules are disabled (e.g., and avoid adding non-compliant code in the future). It also helps to clarify which lines actually triggers a warning.

This applies to a lot of other places in this PR as well.

**Review by bauer-andreas — DISMISSED** [review]
Thank you for your work 🚀 

I just had a small suggestion on how we could `eslint-disable-next` comments to have a better overview of what triggered the warning and avoid disabling "too much". 👍

File: src/internal-types.ts (hunk 1 above)
Comment (bwp91) [reply src/internal-types.ts]: Hi @Supereg yep of course have just updated the PR.

I don't mind either way having the eslint disable rules more specific to the lines they ignore, my initial preference was just to not split the class definition line from the docs above it

https://github.com/homebridge/HAP-NodeJS/blob/95735a759c7f0ba240145a383830bffe4078a406/src/lib/Service.ts#L176-L182

**Review by donavanbecker — APPROVED** [review]

File: src/internal-types.ts (hunk 1 above)
Comment (bauer-andreas) [reply src/internal-types.ts]: There is also `// eslint-disable-line ...` if you want to put the comment into the same line as the warning.

The documentation comment should still attach to the type or is it now dangling and breaks docs generation?

**Review by bauer-andreas — APPROVED** [review]

**Review by donavanbecker — APPROVED** [review]

File: src/internal-types.ts (hunk 1 above)
Comment (bwp91) [reply src/internal-types.ts]: the docs are still generating fine

**Pull request merged**